client/src/modules/dashboard/Events/components/EventCancelButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { Button, ButtonProps } from '@chakra-ui/react';
+import { useConfirm } from 'chakra-confirm';
+import React, { useMemo } from 'react';
+import { Event, useCancelEventMutation } from '../../../../generated/graphql';
+import { EVENT, EVENTS } from '../graphql/queries';
+
+interface EventCancelButtonProps extends ButtonProps {
+  isFullWidth?: boolean;
+  buttonText: string;
+  event: Pick<Event, 'id' | 'canceled'>;
+}
+
+const EventCancelButton: React.FC<EventCancelButtonProps> = (props) => {
+  const { isFullWidth, buttonText, event, ...rest } = props;
+
+  const [cancel] = useCancelEventMutation();
+
+  const confirmCancel = useConfirm({
+    title: 'Are you sure you want to cancel this',
+    body: 'Canceling this will send emails to everyone who RSVPd',
+    buttonColor: 'orange',
+  });
+
+  const data = useMemo(
+    () => ({
+      variables: { id: event.id },
+      refetchQueries: [
+        { query: EVENT, variables: { id: event.id } },
+        { query: EVENTS },
+      ],
+    }),
+    [event],
+  );
```

Comment:
I know this is a pattern we use elsewhere, but I don't think `useMemo` helps. Creating the `data` object isn't expensive.  Fetching `event` is, but that's already happened by this point.

There may be some optimisation we *can* do, but I don't think this is it.